src/Shared/HttpSys/RequestProcessing/RequestHeaders.cs

```diff
@@ -171,7 +171,7 @@ public bool TryGetValue(string key, out StringValues value)
                         throw new ArgumentOutOfRangeException("value", value.Value, "Cannot be negative.");
                     }
                     _contentLengthText = HeaderUtilities.FormatNonNegativeInt64(value.Value);
-                    this[HttpKnownHeaderNames.ContentLength] = _contentLengthText;
+                    ((IHeaderDictionary)this)[HttpKnownHeaderNames.ContentLength] = _contentLengthText;
```

Comment:
These casts can be reverted that you've fixed the indexer, right?

Reply:
Yep. My first think was that inside explicit implementation of `IHeaderDictionary.ContentLength` we should use explicit calls, but now i think that its doesn't mater in this case. Thanks.

Rolling back, did I get that right?

Reply:
There is only one indexer now, there's no need to use the IHeaderDictionary prefix. Yes, please remove it.